### ISSUE TITLE
FR - Add attributes for styling options

### ISSUE THREAD

**Issue by ScottKillen** [marcusolsson/obsidian-projects#560]
### What would you like to be added?

I would like to see meta-data attributes added to help with styling, in the same vein as recent obsidian versions and other plugins.

For example. In the table, if I am showing a field (e.g. waiting), the code for that field is:

```
<div class="setting-item-description" style="margin-bottom: 4px;">waiting</div>
```

What I would like would be something like:

```
<div data-key="waiting" class="setting-item-description" style="margin-bottom: 4px;">waiting</div>
```

Additionally, it is hard to style field values because only the temporal svelte class is applied.

### Why is this needed?

I would like to be able to style specific values. This would update the plugin to the current obsidian standard and also help it compete better with other plugins that provide this functionality (e.g. [kanban](https://github.com/mgmeyers/obsidian-kanban))

**Comment by marcusolsson** [comment]
I don't have any plans to prioritize theming capabilities at the moment. The UI is still subject to change, and I don't want to make any promises until I can actively maintain a theming API. I'll likely move away from the auto-generated Svelte classes, but for now they serve as a hint not to rely on any class names.

TL;DR I don't want to make promises I can't keep.
